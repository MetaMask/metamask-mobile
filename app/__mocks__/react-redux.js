export * from 'react-redux-test';
