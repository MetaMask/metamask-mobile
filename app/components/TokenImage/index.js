import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import AssetIcon from '../AssetIcon';
import Identicon from '../Identicon';

const styles = StyleSheet.create({
	itemLogoWrapper: {
		width: 50,
		height: 50,
		overflow: 'hidden',
		borderRadius: 100,
		marginRight: 20
	}
});

/**
 * View that renders an ERC-20 Token logo
 */
export default class TokenElement extends Component {
	static propTypes = {
		/**
		 * Asset object (in this case ERC20 token)
		 */
		asset: PropTypes.object
	};

	render = () => {
		const { asset } = this.props;
		return (
			<View style={styles.itemLogoWrapper}>
				{asset.logo ? <AssetIcon logo={asset.logo} /> : <Identicon address={asset.address} />}
			</View>
		);
	};
}
