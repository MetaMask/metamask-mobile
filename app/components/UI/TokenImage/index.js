import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import AssetIcon from '../AssetIcon';
import Identicon from '../Identicon';
import isUrl from 'is-url';
import { connect } from 'react-redux';
import { getTokenList } from '../../../reducers/tokens';

const styles = StyleSheet.create({
  itemLogoWrapper: {
    width: 50,
    height: 50,
  },
  roundImage: {
    overflow: 'hidden',
    borderRadius: 25,
  },
});

const TokenImage = ({ asset, containerStyle, iconStyle, tokenList }) => {
  const assetImage = isUrl(asset?.image) ? asset.image : null;
  const iconUrl =
    assetImage ||
    tokenList[asset?.address]?.iconUrl ||
    tokenList[asset?.address?.toLowerCase()]?.iconUrl ||
    '';

  return (
    <View style={[styles.itemLogoWrapper, containerStyle, styles.roundImage]}>
      {iconUrl ? (
        <AssetIcon
          address={asset?.address}
          logo={iconUrl}
          customStyle={iconStyle}
        />
      ) : (
        <Identicon address={asset?.address} customStyle={iconStyle} />
      )}
    </View>
  );
};

TokenImage.propTypes = {
  asset: PropTypes.object,
  containerStyle: PropTypes.object,
  iconStyle: PropTypes.object,
  tokenList: PropTypes.object,
};

const mapStateToProps = (state) => ({
  tokenList: getTokenList(state),
});

export default connect(mapStateToProps)(TokenImage);
