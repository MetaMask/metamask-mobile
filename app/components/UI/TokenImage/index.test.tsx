import React from 'react';
import { shallow } from 'enzyme';
import TokenImage from './';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore();
const initialState = {
  engine: {
    backgroundState: {
      SwapsController: {
        tokens: [],
      },
      TokenListController: {
        tokenList: {},
      },
    },
  },
  settings: {
    primaryCurrency: 'usd',
  },
};
const store = mockStore(initialState);

describe('TokenImage', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <TokenImage
          asset={{
            address: '0x123',
            symbol: 'ABC',
            decimals: 18,
            image: 'invalid-uri',
          }}
        />
      </Provider>,
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
