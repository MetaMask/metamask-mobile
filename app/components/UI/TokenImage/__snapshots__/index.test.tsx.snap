// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenImage should render correctly 1`] = `
<TokenImage
  asset={
    Object {
      "address": "0x123",
      "decimals": 18,
      "image": "invalid-uri",
      "symbol": "ABC",
    }
  }
  dispatch={[Function]}
  tokenList={Object {}}
/>
`;
